packages/image_picker/image_picker_macos/macos/image_picker_macos/Sources/image_picker_macos/ImagePickerImpl.swift

```diff
@@ -0,0 +1,519 @@
+          .failure(
+            PigeonError(
+              code: "WINDOW_NOT_FOUND", message: "No active window to present the picker.",
+              details: nil)))
+      })
+  }
+
+  func pickVideos(
+    generalOptions: GeneralOptions, completion: @escaping (Result<[String], any Error>) -> Void
+  ) {
+    guard #available(macOS 13.0, *) else {
+      completion(
+        .failure(
+          PigeonError(
+            code: "UNSUPPORTED_PHPICKER",
+            message:
+              "PHPicker is only supported on macOS 13.0 or newer. Use \(supportsPHPicker) to check.",
+            details: nil)))
+      return
+    }
+
+    if generalOptions.limit != nil && generalOptions.limit != 1 {
+      completion(
+        .failure(
+          PigeonError(
+            code: "UNIMPLEMENTED", message: "Multi-video selection is not implemented", details: nil
+          )))
+      return
+    }
+
+    var config = PHPickerConfiguration()
+    config.selectionLimit = 1
+    config.filter = .videos
+
+    let picker = PHPickerViewController(configuration: config)
+    pickVideosDelegate = PickVideosDelegate(completion: completion)
+    picker.delegate = pickVideosDelegate
+
+    showPHPicker(
+      picker,
+      noActiveWindow: {
+        completion(
+          .failure(
+            PigeonError(
+              code: "WINDOW_NOT_FOUND", message: "No active window to present the picker.",
+              details: nil)))
+      })
+  }
+
+  func pickMedia(
+    options: MediaSelectionOptions, generalOptions: GeneralOptions,
+    completion: @escaping (Result<[String], any Error>) -> Void
+  ) {
+    guard #available(macOS 13.0, *) else {
+      completion(
+        .failure(
+          PigeonError(
+            code: "UNSUPPORTED_PHPICKER",
+            message:
+              "PHPicker is only supported on macOS 13.0 or newer. Use \(supportsPHPicker) to check.",
+            details: nil)))
+      return
+    }
+
+    var config = PHPickerConfiguration()
+    config.selectionLimit = Int(generalOptions.limit)
+    config.filter = PHPickerFilter.any(of: [.images, .videos])
+
+    let picker = PHPickerViewController(configuration: config)
+    pickMediaDelegate = PickMediaDelegate(completion: completion, options: options)
+    picker.delegate = pickMediaDelegate
+
+    showPHPicker(
+      picker,
+      noActiveWindow: {
+        completion(
+          .failure(
+            PigeonError(
+              code: "WINDOW_NOT_FOUND", message: "No active window to present the picker.",
+              details: nil)))
+      })
+  }
+
+  @available(macOS 13, *)
+  private func showPHPicker(_ picker: PHPickerViewController, noActiveWindow: @escaping () -> Void)
+  {
+    guard let window = NSApplication.shared.keyWindow else {
+      noActiveWindow()
+      return
+    }
+    // TODO(EchoEllet): IMPORTANT The window size of the picker is smaller than expected, see the video in https://discord.com/channels/608014603317936148/1295165633931120642/1295470850283147335
```

Comment:
This is an important issue to fix since it affects the user experience, see the video in this PR.